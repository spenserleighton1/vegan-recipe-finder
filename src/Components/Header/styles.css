.header {
  background-color: rgba(88, 121, 36, 0.8);
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  height: 4rem;
  margin: 0px;
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
}

body {
  font-family: 'Cinzel', serif;
}

h1 {
  position: absolute;
  left: 1.5rem;
  margin: .25rem;
}

.header-link {
  text-decoration: none;
  color: white;
}

.header-btn {
  position: absolute;
  right: 1.5rem;
  font-size: 16px;
  font-family: 'Cinzel', serif;
  background-color: transparent;
  color: white;
  border: none;
  font-size: 
}



