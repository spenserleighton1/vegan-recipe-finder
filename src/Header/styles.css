.header {
  background-color: rgba(88, 121, 36, 0.8);
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  height:50px;
  margin: 0px;
}

body {
  font-family: 'Cinzel', serif;
}

h1 {
  margin: 0;
}

.header-link {
  text-decoration: none;
  color: white;
}

.header-btn {
  font-size: 16px;
  font-family: 'Cinzel', serif;
  background-color: transparent;
  color: white;
  border: none;
  font-size: 
}



