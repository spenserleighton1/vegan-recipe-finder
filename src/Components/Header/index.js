import React from 'react';
import { auth } from '../../firebase';
import './styles.css';
import { Link } from 'react-router-dom';
import Search from '../../Containers/Search';
import PropTypes from 'prop-types';

const Header = ({ authUser }) => {
  return (
    <div>
      { authUser ? <HeaderAuth /> : <HeaderNonAuth /> } 
    </div>
  );
};

const HeaderAuth = () => {
  return (
    <div className='header'>
      <h1>Vegan Recipe Finder</h1>
      <Search />
      <button className='header-btn'>
        <Link className='header-link' exact to={'/savedRecipes'}>Saved Recipes</Link>
      </button>
      <button className='header-btn' type='button' onClick={ () => { auth.doSignOut(); }}>
        <Link className='header-link' to={'/'}>Sign Out</Link>
      </button>
    </div>
  );
};

const HeaderNonAuth = () => { 
  return (
    <div className='header'>
      <Search />
      <h1>Vegan Recipe Finder</h1>
      <button className='header-btn'>
        <Link className='header-link' to={'/signIn'}>Sign In</Link>
      </button>
    </div>
  );
};

export default Header;

Header.propTypes = {
  authUser: PropTypes.object
};

