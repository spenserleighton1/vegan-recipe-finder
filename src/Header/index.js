import React from 'react';
import { auth } from '../firebase';
import './styles.css';
import { Link } from 'react-router-dom';

const Header = ({ authUser }) => {
  return (
    <div>
      { authUser ? <HeaderAuth /> : <HeaderNonAuth /> } 
    </div>
  )
}

const HeaderAuth = () => {
  return (
  <div className='header'>
    <h1>Vegan Recipe Finder</h1>
      <button><Link className='header-link' to={'/savedRecipes'}>Saved Recipes</Link></button>
      <button className='header-btn' type='button' onClick={ () => {auth.doSignOut()}}>Sign Out</button>
  </div>
    )
}

const HeaderNonAuth = () => { 
  return (
  <div className='header'>
    <h1>Vegan Recipe Finder</h1>
    <button className='header-btn'><Link className='header-link' to={'/signIn'}>Sign In</Link></button>
  </div>
    )
}

export default Header;


