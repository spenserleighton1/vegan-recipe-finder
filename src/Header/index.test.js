import React from 'react';
import { shallow } from 'enzyme';
import Header from './'

describe('Header', () => {
  it('should match the snapshot', () => {
    const wrapper = (<Header />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should render an h1 tag with the Header text', () => {
    const wrapper = (<Header />);

    expect(wrapper.find('.header')).to.have.length(1);
  })
})
