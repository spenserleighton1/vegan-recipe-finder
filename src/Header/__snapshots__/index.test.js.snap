// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Vegan Recipe Finder",
      "className": "header",
    },
    "ref": null,
    "rendered": "Vegan Recipe Finder",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Vegan Recipe Finder",
        "className": "header",
      },
      "ref": null,
      "rendered": "Vegan Recipe Finder",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
